great_expectations/rule_based_profiler/parameter_builder/histogram_parameter_builder.py

```diff
@@ -48,9 +59,10 @@ class PartitionParameterBuilder(MetricSingleBatchParameterBuilder):
     def __init__(
         self,
         name: str,
-        bucketize_data: Union[str, bool] = True,
+        bucketize_data: bool = True,
```

Comment:
@jcampbell The rationale for declaring `bucketize_data: Union[str, bool] = True,` is that the string allows us to retrieve it from `$variables` as is done further down in `_build_parameters()`.  As @billdirks pointed out, this constant appearance of `Union` is inelegant; there is a JIRA item now with the ideas for how to eliminate it, without losing the capabilities.  Thanks.

```
# Obtain bucketize_data directive from "rule state" (i.e., variables and parameters); from instance variable otherwise.
```

Reply:
Ok, I'll revert to using `get_parameter_value_and_validate_return_type` in this PR, but yes please let's address that.

Reply:
I don't understand how that is intended to work; how should one access `bucketize_data` here?